app.R

```diff
@@ -0,0 +1,1268 @@
+                    width = 8,
+                         align = "center",
+                         conditionalPanel(
+                           condition = "input.random == 'Null with equal probabilities'",
+                            tableOutput("equalValues"),
+                           
+                           conditionalPanel(
+                             condition = "input.random == 'Null with equal probabilities'",
+                             plotOutput(outputId = "equalPlot",
+                                        width = '90%',
+                                        click = "plot_click"),
+                             
+                             tableOutput("equalPlotClickedpoints"),
+  
+                             htmlOutput(
+                               outputId = "equalText",
+                               class = "text-center")
+                                          ))),
+
+                  column(
+                    width = 8,
+                         align = "center",
+                    
+                         conditionalPanel(
+                           condition = "input.random == 'Null with different probabilities'",
+                           tableOutput("diffValues"),
+                           
+                           plotOutput(outputId = "diffPlot",
+                                      width = "90%",
+                                      click = "plot_click"),
+                           
+                           tableOutput("diffPlotClickedpoints"),
+                           
+                            htmlOutput(
+                              outputId = "diffText",
+                              class = "text-center")
+                           )))
+        ),
+
+        #### Set up the Explore Page ----
+        tabItem(
+          tabName = "explore",
+          withMathJax(),
+          # Use h2 here
+          h2("Explore Your Own Data"),
+          p("You can use this page to see the behavior of distribution of the chi-square statistic for your data.
+                  If you need guidence for each part, put arrow on that box.
+                  When you finished, click on Simulate Now to see the plots."),
+          fluidRow(
+            #column of length 12 which is the whole width
+            #I include everthing in a column though because this way there are margins and it looks better
+            ##### sideBar on the left ----
+            column(
+                   width = 4,
+                   wellPanel(
+                     align = "left",
+                   textInput(inputId = "names",
+                             label = HTML("Level names", 
+                                          as.character(
+                                            actionLink(inputId = "namesPopover", 
+                                                       label = "", 
+                                                       icon = icon("info")))),
```

Comment:
These icons contain important information. However, I think it may be an accessibility issues. I think you could add this information right into the instructions / description at the top. What do you think?

Reply:
Hi Shravani,
Thank you for pointing it out. I and Dr.Pearl had also talked about how to deal with these hints and we eventually decided to make them into this form. Therefore, I think we can just leave it like this.